sentry-core/src/main/java/io/sentry/core/SentryEnvelope.java

```diff
@@ -33,25 +34,47 @@ public SentryEnvelope(
   }
 
   public SentryEnvelope(
-      final @Nullable SentryId eventId, final @NotNull Iterable<SentryEnvelopeItem> items) {
-    header = new SentryEnvelopeHeader(eventId);
+      final @Nullable SentryId eventId,
+      final @Nullable SdkInfo sdkInfo,
+      final @NotNull Iterable<SentryEnvelopeItem> items) {
+    header = new SentryEnvelopeHeader(eventId, sdkInfo);
     this.items = Objects.requireNonNull(items, "SentryEnvelope items are required.");
   }
 
-  public SentryEnvelope(final @Nullable SentryId eventId, final @NotNull SentryEnvelopeItem item) {
+  public SentryEnvelope(
+      final @Nullable SentryId eventId, final @NotNull Iterable<SentryEnvelopeItem> items) {
+    this(eventId, null, items);
+  }
+
+  public SentryEnvelope(
+      final @Nullable SentryId eventId,
+      final @Nullable SdkInfo sdkInfo,
+      final @NotNull SentryEnvelopeItem item) {
     Objects.requireNonNull(item, "SentryEnvelopeItem is required.");
 
-    header = new SentryEnvelopeHeader(eventId);
+    header = new SentryEnvelopeHeader(eventId, sdkInfo);
     final List<SentryEnvelopeItem> items = new ArrayList<>(1);
     items.add(item);
     this.items = items;
   }
 
+  public SentryEnvelope(final @Nullable SentryId eventId, final @NotNull SentryEnvelopeItem item) {
```

Comment:
M: Can we get rid of this overload to avoid creating envelopes without sdk-info?